sql-plugin/src/main/scala/com/nvidia/spark/rapids/JoinGatherer.scala

```diff
@@ -670,4 +677,137 @@ case class MultiJoinGather(left: JoinGatherer, right: JoinGatherer) extends Join
   }
 
   override def toString: String = s"MULTI-GATHER $left and $right"
-}
\ No newline at end of file
+}
+
+/**
+ * JoinGatherer for existence join
+ * Existence join generates an `exists` boolean column with `true` or `false` in it,
+ * then appends it to the `output` columns. It does not shrink or expand left table.
+ * Just provides the `exists` column for the next following operator(usually filter).
+ *
+ * e.g.:
+ * <code>
+ * select * from left_table where
+ *   left_table.column_0 >= 3
+ *   or
+ *   exists (select * from right_table where left_table.column_1 < right_table.column_1)
+ *
+ * Explanation of this sql is:
+ *
+ * Filter(left_table.column_0 >= 3 or `exists`)
+ *   Existence_join // generate `exists` column, do not shrink or expand the rows of left table
+ *     left_table
+ *     right_table
+ * </code>
+ */
+class ExistenceJoinGathererImpl(
+    private val gatherMap: LazySpillableGatherMap,
+    private val data: LazySpillableColumnarBatch,
```

Comment:
IMO it makes no sense to mark these as `val` when they become "useless" immediately upon construction due to the creation of `lazyLeftTableWithExists`.

Reply:
Updated